### PR TITLE
Adding ISSUE_TEMPLATE

### PR DESCRIPTION
**Pull request by blackdwarf**
Addin template for new issues. Also adding instructions for "clone, run, repro" that we want to be at.

/cc @piotrpMSFT @brthor

### CODE REVIEW

File: ISSUE_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Expected  behavior
+
+
+## Actual behavior
+
+
+## Steps to reproduce
```
Comment (livarcocc) [inline ISSUE_TEMPLATE]: I would move the repro steps to be beginning, I think it makes a more natural sense and tells the story better.

Comment (blackdwarf) [inline ISSUE_TEMPLATE]: Really? Most of the templates I saw in life have this form. But, I'm fine either way of course. :)

File: Documentation/issue-filing-guide.md [hunk 2]
```diff
@@ -37,3 +37,15 @@ you would like to see, what scenario it covers and especially why the current
 tools cannot satisfy that scenario. 
 
 Thanks and happy filing! :)
+
+## Providing the repro for bugs
+For bugs, what we need more than anything is a good repro of the defective 
+behavior. We would like to go towards the "clone, run, repro" model. In short:
+
+1. If you find a bug, package up a repro in a git repo somewhere (GitHub is 
+usually a good place :)).
```
Comment (brthor) [inline Documentation/issue-filing-guide.md]: What about `:smile:`? Or do you prefer the ascii emoji

Comment (blackdwarf) [inline Documentation/issue-filing-guide.md]: ASCII all the way dude! :)

Comment (brthor) [inline Documentation/issue-filing-guide.md]: <|:^)

**Pull request merged**